### PR TITLE
Add spouse review page, spouse remove confirmation page and remove spouse functionality

### PR DESCRIPTION
**Pull request by embarnard**
Co-authored-by: Ben Golder <bgolder@codeforamerica.org>

### CODE REVIEW

File: app/controllers/ctc/questions/spouse_info_controller.rb [hunk 1]
```diff
@@ -6,6 +6,10 @@ class SpouseInfoController < QuestionsController
 
       layout "intake"
 
+      def edit
```
Comment (embarnard) [inline app/controllers/ctc/questions/spouse_info_controller.rb:9]: Remove this

File: app/assets/stylesheets/components/_review-box.scss [hunk 2]
```diff
@@ -25,4 +25,14 @@
...
     }
   }
-}
\ No newline at end of file
+
+  &__button {
+    border: 2px solid black;
+    line-height: 2px;
+  }
+}
```
Comment (bengolder) [inline app/assets/stylesheets/components/_review-box.scss:33]: We might check with Anu to see if this is a new consistent button type, and refactor it into the buttons css file

File: app/controllers/ctc/questions/remove_spouse_controller.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+module Ctc
+  module Questions
+    class RemoveSpouseController < QuestionsController
+      include AuthenticatedCtcClientConcern
+      include AnonymousIntakeConcern
+
+      layout "intake"
```
Comment (bengolder) [inline app/controllers/ctc/questions/remove_spouse_controller.rb:7]: Do all the controllers in the CTC flow use the same layout? Could this be inherited?

File: app/forms/ctc/remove_spouse_form.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+module Ctc
+  class RemoveSpouseForm < QuestionsForm
+    set_attributes_for :intake,
+                       :spouse_first_name,
+                       :spouse_middle_initial,
+                       :spouse_last_name,
+                       :spouse_tin_type,
+                       :spouse_ssn,
+                       :spouse_veteran,
+                       :spouse_birth_date
```
Comment (bengolder) [inline app/forms/ctc/remove_spouse_form.rb:10]: Not sure that this is needed for this particular form

File: app/forms/ctc/spouse_info_form.rb [hunk 5]
```diff
@@ -34,6 +34,25 @@ def save
...
+      }
+    end
+
+    def self.date_of_birth_attributes(intake)
+      {
+        spouse_birth_date_day: intake.spouse_birth_date&.day,
+        spouse_birth_date_month: intake.spouse_birth_date&.month,
+        spouse_birth_date_year: intake.spouse_birth_date&.year
+      }
+    end
```
Comment (bengolder) [inline app/forms/ctc/spouse_info_form.rb:54]: maybe write a simple spec for this

**Pull request merged**